messages.yaml

```diff
@@ -0,0 +1,1109 @@
+LintCode:
+  avoid_empty_else:
+    documentation: |-
+      #### Description
+
+      The analyzer produces this diagnostic when the statement after an `else`
+      is an empty statement (a semicolon).
+
+      #### Example
+
+      The following code produces this diagnostic because the statement
+      following the `else` is an empty statement:
+
+      ```dart
+      void f() {
+        if (x > y)
+          print("1");
+        else [!;!]
+          print("2");
+      }
+      ```
+
+      #### Common fixes
+
+      If the statement after the empty statement is intended to be executed only
+      when the condition is `false`, then remove the empty statement:
+
+      ```dart
+      void f() {
+        if (x > y)
+          print("1");
+        else
+          print("2");
+      }
+      ```
+
+      If there is no code that is intended to be executed only when the
+      condition is `false`, then remove the whole `else` clause:
+
+      ```dart
+      void f() {
+        if (x > y)
+          print("1");
+        print("2");
+      }
+      ```
+    justification: |-
+      It's easy to miss the presence of the semicolon, which can make it hard to
+      spot bugs in the code control flow.
+    seeAlso: |-
+      The `empty_statements` lint catches this case and other similar cases.
+
+      The `curly_braces_in_flow_control_structures` lint also prevents this
+      problem by forcing the statement after the `else` to be a block.
```

Comment:
I really like this. If we do break up the categories, perhaps they should also go under documentation.